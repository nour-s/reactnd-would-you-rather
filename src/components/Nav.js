import React from "react";
import { NavLink } from "react-router-dom";

export default class Nav extends React.Component {
	render() {
		return (
			<React.Fragment>
				<nav className="nav">
					<ul>
						<li>
							<NavLink to="/" exact activeClassName="active">
								Home
							</NavLink>
						</li>
						<li>
							<NavLink to="/add" exact activeClassName="active">
								New Question
							</NavLink>
						</li>
						<li>
							<NavLink
								to="/leaderboard"
								exact
								activeClassName="active"
							>
								Leader Board
							</NavLink>
						</li>
					</ul>
				</nav>
			</React.Fragment>
		);
	}
}
