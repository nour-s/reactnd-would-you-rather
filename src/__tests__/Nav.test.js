import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Nav from "../components/Nav";

describe("Nav", () => {
	it("Render Nav without crashing", () => {
		const renderer = new ShallowRenderer();
		renderer.render(<Nav />);
		const tree = renderer.getRenderOutput();
		expect(tree).toMatchSnapshot();
	});
});
