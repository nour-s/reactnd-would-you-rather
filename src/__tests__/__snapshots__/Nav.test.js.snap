// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Render Nav without crashing 1`] = `
<React.Fragment>
  <nav
    className="nav"
  >
    <ul>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          exact={true}
          to="/add"
        >
          New Question
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          exact={true}
          to="/leaderboard"
        >
          Leader Board
        </NavLink>
      </li>
    </ul>
  </nav>
</React.Fragment>
`;
